charts/spire/charts/spire-server/templates/post-install-hook.yaml

```diff
@@ -0,0 +1,66 @@
+{{- if eq (.Values.controllerManager.enabled | toString) "true" }}
```

Comment:
why? If the workaround works for now, then that is the fix until the controller manager is fixed.

I don't like to have tons and tons of options.

It either works or it doesn't. Havong a value for every single bit is a bit overkill I think.

Reply:
As previously discussed, the workaround always runs, whether the chart runs or not. It can cause some issues with ci/cd systems. I do not want it running every time I do a helm upgrade. It needs an option to disable it. helm has a flag to skip hooks, but since the resource itself is forcing a change, then the flag doesnt work properly either. So a value is needed.

Reply:
I wrapped the hook in `{{- if eq .Values.controllerManager.validatingWebhookConfiguration.failurePolicy "Fail" }}` so they won't run if the FailurePolicy is Ignore. That should cover your use case without adding more flags.

Reply:
Especially in CD I need to run it. Without this fix the upgrade just fails. Still don't get the point on having this configurable? I guess I need to take some time testing to understand this part.

Reply:
@faisal-memon ah, thats cleaver. +1.